code/ui/ui_main.c

```diff
@@ -263,19 +265,39 @@ void UI_WideScreenMode(qboolean on) {
 UI_UpdateWidescreen
 =================
 */
-extern vmCvar_t ui_widescreen;
 static void UI_UpdateWidescreen(void) {
```

Comment:
The idea of UI_UpdateWidescreeen() was to encapsulate whole strategy/conditional logic inherent to implementing widescreen aspect correction in it. All the conditional logic is in UI_UpdateWidescreen() and results of this logic are in non-conditional uiInfo.screenWidth and screenHeight (screenXFactor and screenXFactorInv are just cached values for optimization, perhaps they shouldn't be even added). This is how it worked so far, it makes it isolated, modular, easy to integrate with existing mods and does not trash already cluttered cgame/ui code with additional widescreen logic. It was violated earlier and I put some effort to fix it in "remove ui_shared.c dependencies" commit. I'm not convinced that there is a need of uiInfo.portraitMode to be used outside of UI_UpdateWidescreen().